### PR TITLE
Modified simulate function for NodeShortCircuit to give only one solution object back

### PR DESCRIPTION
**Pull request by SabineAuer**

### CODE REVIEW

File: src/simulations/NodeShortCircuit.jl [hunk 1]
```diff
@@ -43,24 +42,31 @@ simulate(nsc::NodeShortCircuit, powergrid, x1, timespan)
...
-    sol3 = solve(prob3, Rodas4(autodiff=false))
+    function regularState(integrator)
+        sol2 = integrator.sol
+        x3 = find_valid_initial_condition(nsc_powergrid, sol2[end]) # Jump the state to be valid for the new system.
+        integrator.f = rhs(powergrid)
+        integrator.u = x3
+    end
 
-    sol1, sol2, sol3
+    cb1 = DiscreteCallback(((u,t,integrator) -> t in nsc.tspan_fault[1]), errorState)
```
Comment (FHell) [inline src/simulations/NodeShortCircuit.jl]: Why is this written like that? For repeated faults? Might be worth a comment. Also, is the solver guaranteed to always find the t where the Callback is true?

Comment (FHell) [reply src/simulations/NodeShortCircuit.jl]: Meaning:

`t in nsc.tspan_fault[1]`

rather than

`t == nsc.tspan_fault[1]`

Comment (FHell) [reply src/simulations/NodeShortCircuit.jl]: Ok, understood it now. The precise timesteps are added as tstops to solve.

Comment (SabineAuer) [reply src/simulations/NodeShortCircuit.jl]: I would like to merge into master then to be able to use this new nsc implementation, ok?

**Pull request merged**